plugin/auth_examples/auth_0x0100.c

```diff
@@ -24,39 +24,17 @@
   It intentionally uses no constants like CR_OK ok PASSWORD_USED_YES.
 */
 
-#include <mysql/plugin.h>
 #include <string.h>
 #include <stdio.h>
 #include <stdlib.h>
 
-#if 0
 #include <mysql/plugin_auth.h>
-#else
+#undef  MYSQL_AUTHENTICATION_INTERFACE_VERSION
 #define MYSQL_AUTHENTICATION_INTERFACE_VERSION 0x0100
-typedef void MYSQL_PLUGIN_VIO; /* we don't use it here */
 
-typedef struct st_mysql_server_auth_info
-{
-  char *user_name;
-  unsigned int user_name_length;
-  const char *auth_string;
-  unsigned long auth_string_length;
-  char authenticated_as[49]; 
-  char external_user[512];
-  int  password_used;
-  const char *host_or_ip;
-  unsigned int host_or_ip_length;
-} MYSQL_SERVER_AUTH_INFO;
-
-struct st_mysql_auth
-{
-  int interface_version;
-  const char *client_auth_plugin;
-  int (*authenticate_user)(MYSQL_PLUGIN_VIO *vio, MYSQL_SERVER_AUTH_INFO *info);
-};
-#endif
+#include <sql_plugin_compat.h>
```

Comment:
Eh, no. The whole point of this exercise was to test a plugin compiled with 0x0100 version of the API, it must not include anything beyond that, not anything from sql and definitelyl not a new plugin_auth.h.

it embeds definitions as they were at the version of 0x0100.